frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/LogsTab.tsx

```diff
@@ -0,0 +1,245 @@
+import EllipsisVIcon from '@patternfly/react-icons/dist/esm/icons/ellipsis-v-icon';
+import DownloadIcon from '@patternfly/react-icons/dist/esm/icons/download-icon';
+import { LogViewer } from '@patternfly/react-log-viewer';
+import { PipelineRunTaskDetails } from '~/concepts/pipelines/content/types';
+import SimpleDropdownSelect from '~/components/SimpleDropdownSelect';
+import useFetchLogs from '~/concepts/k8s/pods/useFetchLogs';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import usePodContainers from '~/concepts/k8s/pods/usePodContainers';
+import { PodContainer } from '~/types';
+
+type LogsTabProps = {
+  task: PipelineRunTaskDetails;
+  isHidden: boolean;
+};
+
+const LogsTab: React.FC<LogsTabProps> = ({ task, isHidden }) => {
+  const { namespace } = usePipelinesAPI();
+  const podName = task.runDetails?.status.podName ?? '';
+  const podContainers = usePodContainers(namespace, podName);
+  const [selectedContainer, setSelectedContainer] = React.useState<PodContainer | null>(
+    podContainers[0],
+  );
+  const [logs, loaded, error, refresh] = useFetchLogs(podName, selectedContainer?.name ?? '');
+  const [reRenderKey, setReRenderKey] = React.useState(0);
+  let data = 'No logs available';
+  if (logs) {
+    data = !loaded ? 'Loading...' : logs;
+  }
+
+  React.useEffect(() => {
+    const interval = setTimeout(() => refresh(), 3000);
+
+    return () => clearInterval(interval);
+  }, [refresh]);
+
+  React.useEffect(() => {
+    setReRenderKey((key) => key + 1);
+  }, [isHidden]);
+
+  const [isPaused, setIsPaused] = React.useState(false);
+  const [isFullScreen] = React.useState(false);
+  const [itemCount, setItemCount] = React.useState(1);
+  const [currentItemCount, setCurrentItemCount] = React.useState(0);
+  const [timer, setTimer] = React.useState(500);
+  const [selectedData] = React.useState([]);
+  const [buffer, setBuffer] = React.useState([]);
+  const logViewerRef = React.useRef<{ scrollToBottom: () => void }>();
+
+  React.useEffect(() => {
+    setTimer(
+      window.setInterval(() => {
+        setItemCount((itemCount) => itemCount + 1);
+      }, 500),
+    );
+    return () => {
+      window.clearInterval(timer);
+    };
+  }, [timer]);
+
+  React.useEffect(() => {
+    if (itemCount > selectedData.length) {
+      window.clearInterval(timer);
+    } else {
+      setBuffer(selectedData.slice(0, itemCount));
+    }
+  }, [itemCount, selectedData, timer]);
+
+  React.useEffect(() => {
+    if (!isPaused && buffer.length > 0) {
+      setCurrentItemCount(buffer.length);
+      if (logViewerRef && logViewerRef.current) {
+        logViewerRef.current.scrollToBottom();
+      }
+    }
+  }, [isPaused, buffer]);
+
+  const onDownloadClick = () => {
+    const element = document.createElement('a');
+    const dataToDownload = data;
+    const file = new Blob([dataToDownload], { type: 'text/plain' });
+    element.href = URL.createObjectURL(file);
+    element.download = `${podName}.txt`;
```

Comment:
```suggestion
    element.download = `${podName}-${selectedContainer}.txt`;
```
We need something more than just the pod name, otherwise when I download two containers from the same pod, I get two files of the same name.